pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -581,23 +618,20 @@ func configureGitPassword(ctx context.Context, be backend.Backend,
 
 	if sd.DeploymentSettings.SourceContext.Git.GitAuth.BasicAuth.UserName.Value != "" &&
 		!sd.DeploymentSettings.SourceContext.Git.GitAuth.BasicAuth.UserName.Secret {
-		username, err = cmdutil.ReadConsoleWithDefault("Enter the git username",
-			sd.DeploymentSettings.SourceContext.Git.GitAuth.BasicAuth.UserName.Value)
-	} else {
-		username, err = cmdutil.ReadConsole("Enter the git username")
+		username = sd.DeploymentSettings.SourceContext.Git.GitAuth.BasicAuth.UserName.Value
 	}
+
+	username, err = promptForValue(d.Yes, "git username", username, false, ValidateGenericInputNonEmpty, *d.DisplayOptions)
 	if err != nil {
 		return err
 	}
 
-	if password, err = cmdutil.ReadConsoleNoEcho("Enter the git password"); err != nil {
+	password, err = promptForValue(d.Yes, "git password", password, true, ValidateGenericInputNonEmpty, *d.DisplayOptions)
```

Comment:
Not sure if these should be capitalised or not? To my mind if they are starting a line they should be `Git username`, `Git password`, etc. but I could be wrong.

Reply:
same as before, happy to change it but I tried to be consistent with existing prompts

<img width="753" alt="image" src="https://github.com/pulumi/pulumi/assets/5647310/8d0a259e-0fea-466a-8252-9377316a5a80">